### ISSUE TITLE
Some improvements maybe

### ISSUE THREAD

**Issue by SunSea88** [TalkLounge/tiktok-live-guest#1]
Hello,

Regarding this [discussion](https://greasyfork.org/fr/scripts/475697-tiktok-live-guest/discussions/207349) on Greasy Fork, I would like to know if it's a problem or not please ?

After several tests, when I enter in a live with your script it can last 1 minute, 4 minutes, 18 minutes, 32 minutes... but it never stays continuous so you always have to reload the page suddenly 😔

Then, the links to enter in the lives are always https://www.tiktok.com/@livehaf.official/live?enter_from_merge=others_homepage&enter_method=others_photo but not directly https://www.tiktok.com/@livehaf.official/live when you click on the profit photo 😟

In addition, it's possible to move forward or backward on TikTok just like on YouTube even if it's a live ?

Do you think that your script can improve all this please ?

**Comment by TalkLounge** [comment]
> never stays continuous

Never watched a livestream so long to discover this bug, it's fixed now. Thanks for reporting

> links to enter

These query parameters in the url are there for a reason. I could remove them but that would be against the best practice. Do they bother you that much?

> it's possible to move forward or backward on TikTok just like on YouTube

What do you mean by that? Do you mean the miniplayer feature from youtube (Overlaying above the page)?

![grafik](https://github.com/TalkLounge/tiktok-live-guest/assets/26689633/a7a5982d-1073-4687-992c-ae531471a205)


![grafik](https://github.com/TalkLounge/tiktok-live-guest/assets/26689633/864848ea-8c60-4d5f-8eed-b84fa68fc010)

<br>

Or do you mean the feature to go to the next/previous tiktok/tiktok live?

![grafik](https://github.com/TalkLounge/tiktok-live-guest/assets/26689633/ec367fd5-1863-4d60-a0db-a3b93223571b)

**Comment by SunSea88** [comment]
> > links to enter
> 
> These query parameters in the url are there for a reason. I could remove them but that would be against the best practice. Do they bother you that much?
> 

It's because I didn't want to have tracking links from any web site or service if it's possible... 
But if these parameters are really important for your script I understand.

> > it's possible to move forward or backward on TikTok just like on YouTube
> 
> What do you mean by that? Do you mean the miniplayer feature from youtube (Overlaying above the page)?
> 
> ![grafik](https://user-images.githubusercontent.com/26689633/282117987-a7a5982d-1073-4687-992c-ae531471a205.png)
> 
> ![grafik](https://user-images.githubusercontent.com/26689633/282119676-864848ea-8c60-4d5f-8eed-b84fa68fc010.png)
> 
> 
> Or do you mean the feature to go to the next/previous tiktok/tiktok live?
> 
> ![grafik](https://user-images.githubusercontent.com/26689633/282119115-ec367fd5-1863-4d60-a0db-a3b93223571b.png)

I rather meant to say to move forward or backward in the Livestream with the mouse click on the video controls bar or the left-right arrows on TikTok like on YouTube for example...

![2023-11-11_194030 - Copie](https://github.com/TalkLounge/tiktok-live-guest/assets/143971989/4b72e9e3-1e37-4f37-b9ef-8f3ef310348f)

I couldn't do it because it was constantly loading with the previous version 1.0.1, but with the new version 1.1.0 it's works now 👍
Thank you very much ☺️

**Comment by TalkLounge** [comment]
> It's because I didn't want to have tracking links from any web site or service if it's possible...

I understand that, but that's not the purpose of the script. There are other userscripts/extensions for minimize tracking for e.g. [ClearURLs](https://github.com/ClearURLs/Addon#readme) (However, it doesn't remove the ones from tiktok).
Otherwise someone else will open another issue, asking why the script removes tracking parameters, that are not essential for the scripts functionality, which is completely true. A userscript should do it's purpose with minimal effort.

> But if these parameters are really important for your script I understand.

I guess these parameters are only there for tracking, yes. But I'm not that deep into tiktoks code base to be 100% sure about that. The last thing I want is, that in some edge case, the script causes the website to crash.

> but with the new version 1.1.0 it's works now

Perfect.
I will close the issue for now as it seems that everything is working now again. You can of course reopen or comment at every time or create a new issue when another bug appears.

**Comment by SunSea88** [comment]
OK and Thank you very much again for all ☺️

**Comment by SunSea88** [comment]
Hello,

Now that TikTok is forcing us to log in to enter into a live, do you think that it's possible for you to get around that please ?
-> https://www.tiktok.com/login?redirect_url=https%3A%2F%2Fwww.tiktok.com%2F%40livehaf.official%2Flive&lang=en

Thank you again in advance.

**Comment by TalkLounge** [comment]
For me the script still works... Hmm
Whats the Problem? Do you get redirected to the login page when opening https://www.tiktok.com/@livehaf.official/live
Does this also happen in incognito mode or with another browser?

**Comment by SunSea88** [comment]
Hello,

Very sorry for my late reply, because I left it a few days to see if the problem will come back on my side...

So far, TikTok hasn't blocked me to force me to log in like it did the previous time with the URL mentioned above or another...
So, the problem lasted a day...


Otherwise, to respond to your answers :

> Whats the Problem? Do you get redirected to the login page when opening [tiktok.com/@livehaf.official/live](https://www.tiktok.com/@livehaf.official/live)

Yes with the URL mentioned above.

> Does this also happen in incognito mode or with another browser?

Yes on Google Chrome exe and portable with Tampermonkey or Violentmonkey.

**Comment by SunSea88** [comment]
In addition, I would like to know if on your side the tab title for the live is :
-> Connexion | TikTok for https://www.tiktok.com/@livehaf.official/live  like the screenshot shows ?

Or you have rather this tab title for the live :
-> Livehaf (@livehaf.official) Officiel | TikTok or other please ?

![2023-12-29_072316 - Copie](https://github.com/TalkLounge/tiktok-live-guest/assets/143971989/1869cd69-9c56-49e9-82d8-a82b65184827)

**Comment by TalkLounge** [comment]
> So, the problem lasted a day

Let me know when it occurs again


For me the tab title is "Livehaf (@livehaf.official) ist LIVE | TikTok" and as soon the login popup opens (which then gets blocked by the script) it changes to "Anmelden | TikTok"

**Comment by SunSea88** [comment]
Okay, I see and Thank you very much for the explanation 👍
